aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/AWSMobileClient.java

```diff
@@ -1611,6 +1611,9 @@ public Void run() throws Exception {
                     userpoolLL.globalSignOut(globalSignOutRequest);
                 }
                 if (signOutOptions.isInvalidateTokens()) {
+                    if (userpool != null) {
+                        userpool.getCurrentUser().revokeTokens();
+                    }
                     if (hostedUI != null) {
```

Comment:
hostedUI sign out should also revoke the token, is this covered by this code path? ie. if hostedUI is not null, does that mean userpool is also not null

Reply:
Yes, tokens are revoked for hostedUI sign out. `userpool` is alway non null whether hostedUI is null or not.